Makefile

```
@@ -17,12 +18,51 @@ endif
 BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 
 OUTPUT_DIR ?= bundle
-BUNDLE_FLAGS = --output-dir=$(OUTPUT_DIR)
+BUNDLE_EXTRA_FLAGS ?=
 
 # Image URL to use all building/pushing image targets
 IMG ?= ocs-osd-deployer:latest
-# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd:trivialVersions=true"
+
+# USE_IMAGE_DIGESTS defines if images are resolved via tags or digests
+USE_IMAGE_DIGESTS ?= false
+ifeq ($(USE_IMAGE_DIGESTS), true)
+    BUNDLE_EXTRA_FLAGS += --use-image-digests
+endif
+
+BUNDLE_IMGS ?= $(BUNDLE_IMG) 
+CATALOG_IMG ?= $(IMAGE_TAG_BASE)-catalog:v$(VERSION)
+
+ifneq ($(origin CATALOG_BASE_IMG), undefined)
+	FROM_INDEX_OPT := --from-index $(CATALOG_BASE_IMG)
+endif 
+
+LOCALBIN ?= $(CURDIR)/bin
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+## Tool Binaries
+KUSTOMIZE ?= $(LOCALBIN)/kustomize
+CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+ENVTEST ?= $(LOCALBIN)/setup-envtest
+OPERATOR_SDK = $(LOCALBIN)/operator-sdk
+OPM ?= $(LOCALBIN)/opm
+
+## Tool Versions
+KUSTOMIZE_VERSION ?= v3.9.1
+CONTROLLER_TOOLS_VERSION ?= v0.8.0
+ENVTEST_VERSION ?= latest
+OPERATOR_SDK_VERSION ?= v1.21.0
+OPM_VERSION ?= v1.19.1
+
+KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
+
+# ===== Constants ===== #
+
+BUNDLE_FLAGS = --output-dir=$(OUTPUT_DIR)
+BUNDLE_GEN_FLAGS = -q --extra-service-accounts prometheus-k8s --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS) $(BUNDLE_FLAGS)
```

`BUNDLE_FLAGS` is a constant that is used only once, in `BUNDLE_GEN_FLAGS`, and is not overridable. So why do we need it?

```suggestion
BUNDLE_GEN_FLAGS = -q --extra-service-accounts prometheus-k8s --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS) --output-dir=$(OUTPUT_DIR)
```

Also the constant `BUNDLE_GEN_FLAGS` itself is being used only once in the Kustomize build command, so why do we need to break the linear readability of this code and force the reader to jump between this constant and the command just to understand what are the flags?

Can we fold this into the command itself?
```
$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle /
    -q /
    --extra-service-accounts prometheus-k8s /
    --overwrite /
    --version $(VERSION) /
    $(BUNDLE_METADATA_OPTS) /
    --output-dir=$(OUTPUT_DIR) /
    $(BUNDLE_EXTRA_FLAGS)
```

---

- I just wanted a var to hold all the possible static info
- anyways, not an issue to inline, done